tests/pytorch/nightly/hf-llm.libsonnet

```diff
@@ -0,0 +1,147 @@
+        '--per_device_train_batch_size=%d ' % config.paramsOverride.per_device_train_batch_size,
+        '--per_device_eval_batch_size=%d ' % config.paramsOverride.per_device_eval_batch_size,
+        '--do_train',
+        '--do_eval',
+        '--logging_dir=./tensorboard-metrics',
+        '--cache_dir=./cache_dir',
+        '--output_dir=/tmp/test-clm',
+        '--overwrite_output_dir',
+        '--cache_dir=/tmp',
+        '--config_name=%s' % config.paramsOverride.config_name,
+        '--tokenizer_name=gpt2',
+        '--block_size=1024',
+        '--optim=adafactor',
+        '--adafactor=true',
+        '--save_strategy=no',
+        '--logging_strategy=no',
+        '--fsdp=full_shard',
+        '--fsdp_config=examples/pytorch/language-modeling/fsdp_config.json',
+      ],
+    },
+    command: utils.scriptCommand(
+      |||
+        %s
+        %s
+        %s
+      ||| % [
+        command_common,
+        utils.toCommandString(self.paramsOverride.trainCommand),
+        command_copy_metrics,
+      ]
+    ),
+  },
+
+
+  local config_2B = self.config_2B,
+  config_2B:: common.Convergence {
+    modelName: 'hf-gpt2-2b',
+    paramsOverride+:: {
+      config_name: '/home/xl-ml-test/transformers/examples/pytorch/language-modeling/my_config_2.json',
+      per_device_train_batch_size: 8,
+      per_device_eval_batch_size: 8,
+    },
+    metricConfig+: {
+      sourceMap+:: {
+        tensorboard+: {
+          aggregateAssertionsMap+:: {
+            'eval/loss': {
+              FINAL: {
+                fixed_value: {
+                  comparison: 'LESS',
+                  value: 7,
+                },
+                inclusive_bounds: false,
+                wait_for_n_data_points: 0,
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+
+  local tpuVm = self.tpuVm,
+  tpuVm:: common.PyTorchTpuVmMixin {
+    tpuSettings+: {
+      tpuVmExports+: |||
+        cd transformers/
+        export LD_LIBRARY_PATH=/usr/local/lib/
+        export PJRT_DEVICE=TPU_C_API
+        export PT_XLA_DEBUG=0
+        export USE_TORCH=ON
+      |||,
+      tpuVmExtraSetup: |||
+        pip install --upgrade accelerate
+        git clone https://github.com/huggingface/transformers.git
+        cd transformers
+        git checkout ebdb185befaa821304d461ed6aa20a17e4dc3aa2
+        pip install .
+        git log -1
+        pip install datasets evaluate scikit-learn
+        gsutil cp -r gs://cloud-tpu-tpuvm-artifacts/config/xl-ml-test/pytorch/gpt2/my_config_*.json examples/pytorch/language-modeling/
+        gsutil cp gs://cloud-tpu-tpuvm-artifacts/config/xl-ml-test/pytorch/gpt2/fsdp_config.json examples/pytorch/language-modeling/
+      |||,
+    },
+  },
+
+  local v4_8 = self.v4_8,
+  v4_8:: {
+    accelerator: tpus.v4_8,
+  },
+
+  local pjrt = self.pjrt,
+  pjrt:: tpuVm + experimental.PjRt {
+    modelName+: '-pjrt',
+  },
+
+  configs: [
+    gpt2_model + v4_8 + config_2B + timeouts.Hours(1) + pjrt,
```

Comment:
What's the default timeout in PjRt?

Reply:
it doesn't change the default timeout setting

Reply:
I remember the functional test limit is 1 hour, so wondering why we have `timeouts.Hours(1)` here? Or this is a convergence test?

Reply:
It is convergence test, and mostly it ends in 30min for 3 epochs. So 1 hour timer here is to monitor this test's performance regression.

Reply:
Thanks! This sounds good to me.